src/pattern-library/examples/use-click-away-basic.tsx

```diff
@@ -0,0 +1,31 @@
+import { useRef } from 'preact/hooks';
+
+import { useClickAway } from '../../hooks/use-click-away';
+
+export default function App() {
+  const containerRef = useRef<HTMLDivElement | null>(null);
+  const wrapperRef = useRef<HTMLDivElement | null>(null);
+
+  useClickAway(containerRef, e => {
+    if (
+      e.target === wrapperRef.current ||
+      wrapperRef.current?.contains(e.target as HTMLElement)
+    ) {
+      alert('You have clicked away');
+    }
+  });
```

Comment:
In this demo I think it is perhaps unclear why you'd use this hook as opposed to just an `onClick` listener on `wrapperRef`. Something like this where you can click anywhere in the page to trigger the state change would be more obvious:

```ts
export default function App() {
  const [clickedAway, setClickedAway] = useState(false);
  const containerRef = useRef<HTMLDivElement | null>(null);
  useClickAway(containerRef, () => setClickedAway(true));

  return (
    <div
      ref={containerRef}
      className={classnames(
        'rounded-lg h-32 w-full p-2 flex items-center justify-center',
        { 'bg-red-100': clickedAway, 'bg-amber-100': !clickedAway },
      )}
    >
      {clickedAway ? "You clicked outside the area" : "Click outside this area"}
      <Button onClick={() => setClickedAway(false)}>Reset</Button>
    </div>
  );
}
```

Reply:
Thanks! I have used your example.